content/en/docs/concepts/workloads/pods/ephemeral-containers.md

```diff
@@ -76,3 +76,208 @@ you can view processes in other containers.
+                              volumes:
+                                - '*'
+
+```
+
+## RBAC
+
+Disallowing ephemeral containers with RBAC could be an option if the feature is not needed and it is not possible to disable the feature completely. The KEP-277: Ephemeral Containers state the following about using RBAC to disable ephemeral containers.
+
+Cluster administrators will be expected to choose from one of the following mechanisms for restricting usage of ephemeral containers:
+
+Use RBAC to control which users are allowed to access the /ephemeralcontainers subresource.
+Write or use a third-party admission controller to allow or reject Pod updates that modify ephemeral containers based on the content of the update.
+Disable the feature using the EphemeralContainers feature gate.
+RBAC is additive which means that it is not possible to remove permissions from a role. This type of mitigation obviously does not matter if all users a cluster admin, which they should not be, so we assume that new roles are created for the cluster consumers. In this case having a look at the existing roles can be enough to make sure that the subresource /ephemeralcontainers is not included in the role.
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: edit
+  rules:
+  - apiGroups:
+    - ""
+      resources:
+        - pods
+          - pods/attach
+            - pods/exec
+              - pods/portforward
+                - pods/proxy
+                  verbs:
+                    - create
+                    - delete
+                    - deletecollection
+                    - patch
+                    - update
+
+```
+## Checking Policy Enforcement
+
+Let's say that you upgraded your cluster and informed all end users of the great new feature. How do you know that the correct policies are enforced in accordance to your security practices. You may have been aware of the API changes and taken the correct precautionary steps. Or you just updated Kyverno and it's policies out of pure happenstance. Either way it is good to trust but verify that it is not for example possible to create a privileged ephemeral container. Annoyingly the debug command does not expose any options to set any security context configuration, so we need another option. Ephemeral containers cannot be defined in a Pod when it is created and it can neither be added with an update. We need some other method to create these specific ephemeral containers.
+
+Ephemeral containers are created using a special ephemeralcontainers handler in the API rather than by adding them directly to pod.spec, so it's not possible to add an ephemeral container using kubectl edit.
+
+The simplest method to add an ephemeral container with a security context to a Pod is to use the Go client. A couple of lines of code can add a new ephemeral container running as privileged or use any other security context setting which is to your liking
+```
+package main
+
+import (
```

Comment:
Is this Go program helpful to understand the concept of ephemeral containers?
I don't get the point.